src/groupeddataframe/splitapplycombine.jl

```diff
@@ -609,42 +611,17 @@ function combine_helper(f, gd::GroupedDataFrame,
             return GroupedDataFrame(newparent, gd.cols, gd.groups, gd.idx,
                                     gd.starts, gd.ends, gd.ngroups, getfield(gd, :keymap))
         else
-            starts = Vector{Int}(undef, length(gd))
-            ends = Vector{Int}(undef, length(gd))
-            starts[1] = 1
-            j = 1
-            for i in 2:length(idx)
-                if idx[i] != idx[i-1]
-                    j += 1
-                    starts[j] = i
-                    ends[j-1] = i - 1
-                end
-            end
-            # it is impossible to get more groups in the output than we had initially
-            @assert j <= length(gd)
-            # In case some groups have to be dropped
-            resize!(starts, j)
-            resize!(ends, j)
-            ends[end] = length(idx)
-
-            groups = zeros(Int, length(idx))
-            for i in 1:j
-                @inbounds for k in starts[i]:ends[i]
-                    groups[k] = i
-                end
-            end
-            # all groups must be filled
-            @assert minimum(groups) == 1
-
+            groups = gen_groups(idx)
+            @assert groups[end] <= length(gd)
             return GroupedDataFrame(newparent, collect(1:length(gd.cols)), groups,
-                                    collect(1:length(idx)), starts, ends, j, nothing)
+                                    nothing, nothing, nothing, groups[end], nothing)
```

Comment:
I was thinking about it. The reason to remove it is the following:
1. we would have to compute them anyway (they are not guaranteed to be given like in the branch above)
2. if we compute them now then we have to store them (so it takes memory), and it is likely that they might be even not needed
3. the cost of computing indices with `compute_indices` is higher than computing them here but not that much higher (it doest 3 loops instead of 2 loops that could be used in an optimal case so the saving is at most ~33%)

For `10^8` groups we have:
```
julia> x = [1:10^8;];

julia> @time DataFrames.compute_indices(x, 10^8);
  1.419073 seconds (8 allocations: 2.235 GiB, 11.12% gc time)
```
and with an optimal timing we would have it a bit over 1 second (at the cost of allocating these 3 vectors - and maybe we even do not have to pay this cost at all if later this `GroupedDataFrame` will be used to do some aggregation function like `sum` when we do not need to pay this cost at all).

In summary:
* if we have to pay this cost we will be slow anyway (as it means we had to use non-aggregating operations which will be more expensive than `compute_indices` anyway)
* if we do not have to pay it it means we are doing aggregations only which are much faster than `compute_indices` (and this is probably the most likely use case)
* additionally we save some memory (this "some" can be a lot if there are a lot of small groups)